src/main/java/thePackmaster/actions/discopack/AnyNumberDiscardAction.java

```diff
@@ -0,0 +1,39 @@
+package thePackmaster.actions.discopack;
+
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.GameActionManager;
+import com.megacrit.cardcrawl.actions.common.GainBlockAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import thePackmaster.SpireAnniversary5Mod;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static thePackmaster.util.Wiz.att;
+import static thePackmaster.util.Wiz.p;
+
+public class AnyNumberDiscardAction extends AbstractGameAction {
+
+    private final List<AbstractCard> selectedCards = new ArrayList<>();
+
+    public AnyNumberDiscardAction(List<AbstractCard> selectedCards) {
+        this.amount = amount;
+        this.selectedCards.addAll(selectedCards);
+    }

```

Comment:
This seems unneeded, just give the SelectCardsInHand action a a lamda with a for each loop that adds discardspecificcardaction on them.

That also will contain any patches made to the discard action.